storey/timescaledb_target.py

```diff
@@ -0,0 +1,322 @@
+    :param columns: list of column names to be written to the hypertable. Will be extracted from events when an event
+        is a dictionary. Use = notation for renaming fields (e.g. write_this=event_field). Use $ notation to refer to
+        metadata ($key, event_time=$time). The time column should not be included in this list as it's specified
+        separately via time_col parameter.
+    :type columns: list[str]
+    :param time_format: If time_col contains string timestamps, this parameter specifies the format for parsing.
+        If not provided, timestamps will be parsed according to ISO-8601 format. Common formats include:
+        "%Y-%m-%d %H:%M:%S", "%d/%m/%y %H:%M:%S UTC%z", etc.
+    :type time_format: Optional[str]
+    :param table: Name of the TimescaleDB hypertable where events will be written. The table must exist and be
+        configured as a hypertable before writing data. If not specified, the table name should be provided through
+        other means (e.g., via batching configuration).
+    :type table: Optional[str]
+    :param max_connections: Maximum number of connections in the asyncpg connection pool. Higher values allow for
+        better concurrency but consume more database resources. Defaults to 10.
+    :type max_connections: int
+    :param min_connections: Minimum number of connections in the asyncpg connection pool. Defaults to 10.
+    :type min_connections: int
+    :param max_events: Maximum number of events to write in a single batch. If None (default), all events will be
+        written on flow termination, or after flush_after_seconds (if flush_after_seconds is set). Larger batches
+        improve write performance but increase memory usage.
+    :type max_events: int
+    :param flush_after_seconds: Maximum number of seconds to hold events before they are written. If None (default),
+        events will be written on flow termination, or after max_events are accumulated (if max_events is set).
+    :type flush_after_seconds: int
+
+    Example:
+        >>> # Basic usage with millisecond precision timestamps
+        >>> target = TimescaleDBTarget(
+        ...     dsn="postgresql://user:pass@localhost:5432/mydb",
+        ...     time_col="timestamp",
+        ...     columns=["sensor_id", "temperature", "humidity"],
+        ...     table="sensor_data",
+        ...     time_format="%Y-%m-%d %H:%M:%S.%f",
+        ...     max_events=1000,
+        ...     flush_after_seconds=5
+        ... )
+
+        >>> # Usage with event metadata
+        >>> target = TimescaleDBTarget(
+        ...     dsn="postgresql://user:pass@localhost:5432/mydb",
+        ...     time_col="event_time",
+        ...     columns=["$key", "value", "source=device_name"],
+        ...     table="events"
+        ... )
+
+    Note:
+        - The target table must be created as a TimescaleDB hypertable before use
+        - The time column must be of TIMESTAMPTZ type for proper time-series functionality
```

Comment:
What's `TIMESTAMPTZ`? It's not mentioned or used anywhere in the code (would expect to see it in the tests if it's really a must as stated here).